lms/djangoapps/learner_recommendations/utils.py

```diff
@@ -214,3 +242,57 @@ def filter_recommended_courses(
             filtered_recommended_courses.append(course_data)
 
     return filtered_recommended_courses
+
+
+def get_programs_based_on_course(request_course, country_code, user):
+    """
+    Returns a program for the course. If a course is part of multiple programs,
+    this function returns the program with the highest price.
+    """
+    max_price, max_price_program = 0, {}
+    programs = get_programs(course=request_course)
+
+    if not programs:
+        return None
+
+    for program in programs:
+        if program.get('status') != 'active' or _has_country_restrictions(program, country_code):
+            continue
+
+        price = program['price_ranges'][0]['total']
+        if price > max_price:
+            if fetch_program_enrollments_by_student(program_uuids=[program.get('uuid')], user=user).exists():
+                continue
+
+            course_keys = [
+                course['key']
+                for course in program.get('courses', [])
+                if course.get('key') and course.get('key') != request_course
+            ]
+            if _remove_user_enrolled_course_keys(user, course_keys):
+                max_price_program = program
+                max_price = price
+
+    if not max_price_program:
+        return None
+
+    course_pacing_type, total_weeks_to_complete = '', 0
+    for courses in max_price_program.get('courses'):
+        for course_run in courses.get('course_runs'):
+            if course_run.get('availability') in ['Current', 'Upcoming']:
```

Comment:
The logic mentioned here doesn't replicate the logic on prospectus.

```suggestion
            if course_run.get('status') == 'published':
```